src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Services/NuGetBrokeredServices.cs

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.ServiceHub.Framework;
+using Microsoft.VisualStudio.Shell.ServiceBroker;
+
+namespace NuGet.PackageManagement.VisualStudio
+{
+    public static class NuGetBrokeredServices
+    {
+        public const string SourceRepositoryProviderServiceName = "NuGet.SourceRepositoryProviderService";
+        public const string SourceRepositoryProviderServiceVersion = "1.0.0";
+
+
+        public static readonly ServiceRpcDescriptor SourceRepositoryProviderService = new ServiceJsonRpcDescriptor(
+              new ServiceMoniker(SourceRepositoryProviderServiceName, new Version(SourceRepositoryProviderServiceVersion)),
+              ServiceJsonRpcDescriptor.Formatters.UTF8,
+              ServiceJsonRpcDescriptor.MessageDelimiters.HttpLikeHeaders);
+
+        public static BrokeredServiceFactory GetSourceRepositoryProviderServiceFactory()
```

Comment:
Is this public member on a public class define din an assembly that is expected to be a public SDK ref assembly? If so, I don't think you'd want to expose the service factory in this way.